src/components/chatroom/SendMessage.tsx

```diff
@@ -0,0 +1,108 @@
+import styled from 'styled-components';
+import useInput from '../../hooks/useInput';
+import { ChatDataType, ChatType } from '../../Interface';
+import dayjs from 'dayjs';
+import { v4 as uuidv4 } from 'uuid';
+import * as api from '../../api/index';
+import { KeyboardEvent, FormEvent, useRef, useEffect } from 'react';
+import { useSetRecoilState } from 'recoil';
+import { chatState } from '../../store/recoil/recoil';
+
+interface SendMessageProps {
+    roomId: number;
+    chatData: ChatDataType[];
+    userId: number;
+}
+
+const SendMessage = ({ roomId, chatData, userId }: SendMessageProps) => {
+    const setChatData = useSetRecoilState(chatState);
+    const { value, onChange, resetValue } = useInput('');
+    const { postMessage } = api;
+    const inputRef = useRef<HTMLTextAreaElement>(null);
+
+    useEffect(() => {
+        if (inputRef.current) {
+            inputRef.current.focus();
+        }
+    }, []);
+
+    const sendMessage = () => {
+        if (value.length !== 0 && value.replace(/ /g, '').length !== 0) {
```

Comment:
```suggestion
        if(/^[\s]*$/.test(value)){
```
현재 코드 상에서는 `shift` + `Enter`로 빈 문자열이 입력이 되는데, 적절한 정규식을 이용해 검사해주시면 길이 0과 공백 모두 검사하실 수 있습니다.